pkg/build/registry/build/strategy.go

```diff
@@ -110,25 +112,39 @@ type detailsStrategy struct {
 
 // Prepares a build for update by only allowing an update to build details.
 // For now, this is the Spec.Revision field
+// We now need to update the build.status.Reason and build.Status.Message
 func (detailsStrategy) PrepareForUpdate(obj, old runtime.Object) {
 	newBuild := obj.(*api.Build)
 	oldBuild := old.(*api.Build)
 	revision := newBuild.Spec.Revision
+	status := newBuild.Status
 	*newBuild = *oldBuild
 	newBuild.Spec.Revision = revision
+	newBuild.Status = status
```

Comment:
@bparees makes sense. So if we are going to allow the update, I agree, we should just allow the Status and Reason through and we can avoid doing the deep copy that @mfojtik mentioned.